internal/runners/shell/shell.go

```diff
@@ -0,0 +1,87 @@
+package shell
+
+import (
+	"github.com/ActiveState/cli/internal/analytics"
+	"github.com/ActiveState/cli/internal/config"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/installation/storage"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/internal/primer"
+	"github.com/ActiveState/cli/internal/runners/activate"
+	"github.com/ActiveState/cli/internal/runners/use"
+	"github.com/ActiveState/cli/internal/subshell"
+	"github.com/ActiveState/cli/internal/virtualenvironment"
+	"github.com/ActiveState/cli/pkg/platform/authentication"
+	"github.com/ActiveState/cli/pkg/platform/model"
+	"github.com/ActiveState/cli/pkg/platform/runtime"
+	"github.com/ActiveState/cli/pkg/platform/runtime/target"
+	"github.com/ActiveState/cli/pkg/project"
+)
+
+type Params struct {
+	Namespace *project.Namespaced
+}
+
+type primeable interface {
+	primer.Auther
+	primer.Prompter
+	primer.Outputer
+	primer.Subsheller
+	primer.Configurer
+	primer.SvcModeler
+	primer.Analyticer
+}
+
+type Shell struct {
+	auth      *authentication.Auth
+	out       output.Outputer
+	svcModel  *model.SvcModel
+	config    *config.Instance
+	subshell  subshell.SubShell
+	analytics analytics.Dispatcher
+}
+
+func NewShell(prime primeable) *Shell {
+	return &Shell{
+		prime.Auth(),
+		prime.Output(),
+		prime.SvcModel(),
+		prime.Config(),
+		prime.Subshell(),
+		prime.Analytics(),
+	}
+}
+
+func (u *Shell) Run(params *Params) error {
+	logging.Debug("Shell %v", params.Namespace)
+
+	projectDir := use.GetLocalProjectPath(params.Namespace, u.config)
+	if projectDir == "" {
+		err := locale.NewInputError("err_use_project_not_checked_out", "", params.Namespace.Project, projectDir)
+		errs.AddTips(err, locale.Tl("use_checkout_first", "", params.Namespace.Project))
```

Comment:
Are these KVs found in the locale file?

Reply:
Yes, they were part of the `state use` improvements ticket.